src/shared/sam/cli/samCliLocalInvoke.ts

```diff
@@ -113,6 +114,22 @@ export class DefaultSamLocalInvokeCommand implements SamLocalInvokeCommand {
             }
         })
 
+        if (params.timer) {
+            await Promise.race([debuggerPromise, params.timer]).then( async (value) => {
+                if (value === false) {
+                    const err = new Error('The SAM process did not make the debugger available within the timelimit')
+                    this.channelLogger.error(
+                        'AWS.samcli.local.invoke.debugger.timeout',
+                        'The SAM process did not make the debugger available within the timelimit',
+                        err
+                    )
+                    await childProcess.kill()
```

Comment:
Did you confirm whether this cleans up the underlying docker container?

Reply:
Doesn't look like SAM CLI does this at this time. Does SAM CLI cleanup on a specific kill signal?